commit a977b2378ff4d88b561d599a1116821c694493f1
Author: David Faure <faure@kde.org>
Date:   Tue Dec 16 18:35:18 2014 +0100

    Don't let the timezone end up as "/etc/localtime" (literally).
    
    This was happening on systems where /usr/share was a symlink (due to lack
    of space) (I realize now that mount --bind would have been much better).
    
    Since the code does canonicalFilePath + startsWith(dir), we need to
    use canonicalFilePath on dir too.

diff --git a/ktimezoned/ktimezoned.cpp b/ktimezoned/ktimezoned.cpp
index f380c09..d92c899 100644
--- a/ktimezoned/ktimezoned.cpp
+++ b/ktimezoned/ktimezoned.cpp
@@ -99,6 +99,7 @@ void KTimeZoned::init(bool restart)
         config.reparseConfiguration();
     KConfigGroup group(&config, "TimeZones");
     mZoneinfoDir     = group.readEntry(ZONEINFO_DIR);
+    mZoneinfoDir     = QFileInfo(mZoneinfoDir).canonicalFilePath(); // just in case /usr or /usr/share is a symlink
     mZoneTab         = group.readEntry(ZONE_TAB);
     mConfigLocalZone = group.readEntry(LOCAL_ZONE);
     QString ztc      = group.readEntry(ZONE_TAB_CACHE, QString());
